src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs

```diff
@@ -0,0 +1,375 @@
+                    }
+                    else if (category)
+                    {
+                        EventLog.WriteEntry(source, message, EventLogEntryType.Warning, (int)eventInstance.InstanceId, (short)eventInstance.CategoryId);
+                        return eventLog.Entries[eventLog.Entries.Count - 1];
+                    }
+                    else
+                    {
+                        EventLog.WriteEntry(source, message, EventLogEntryType.Warning, (int)eventInstance.InstanceId);
+                        return eventLog.Entries[eventLog.Entries.Count - 1];
+                    }
+                }
+                else if (type)
+                {
+                    EventLog.WriteEntry(source, message, EventLogEntryType.Warning);
+                }
+                else
+                {
+                    EventLog.WriteEntry(source, message);
+                }
+
+                return eventLog.Entries[eventLog.Entries.Count - 1];
+            }
+        }
+
+        private EventLogEntry WriteLogEntryEventSource(string source, bool data = false)
+        {
+            if (data)
+                EventLog.WriteEvent(source, eventInstance, rawData, insertStrings);
+            else
+                EventLog.WriteEvent(source, eventInstance, insertStrings);
+
+            using (EventLog eventLog = new EventLog())
+            {
+                eventLog.Source = source;
+                return eventLog.Entries[eventLog.Entries.Count - 1];
+            }
+
+        }
+
+        private EventLogEntry WriteLogEntryEvent(string source, bool data = false)
+        {
+            using (EventLog eventLog = new EventLog())
+            {
+                string[] insertStringsSingleton = { "ExtraText" };
+                eventLog.Source = source;
+                if (data)
+                    eventLog.WriteEvent(eventInstance, rawData, insertStringsSingleton);
+                else
+                    eventLog.WriteEvent(eventInstance, insertStringsSingleton);
+
+                return eventLog.Entries[eventLog.Entries.Count - 1];
+            }
+        }
+
+        [ConditionalTheory(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void WriteEntry(bool sourceFlag)
+        {
+            string log = "Entry";
+            string source = "Source" + nameof(WriteEntry);
+            try
+            {
+                EventLog.CreateEventSource(source, log);
+                EventLogEntry eventLogEntry;
+                if (sourceFlag)
+                    eventLogEntry = WriteLogEntry(source);
+                else
+                    eventLogEntry = WriteLogEntryWithSource(source);
+
+                Assert.Contains(message, eventLogEntry.Message);
+                Assert.Equal(source, eventLogEntry.Source);
+                Assert.StartsWith(Environment.MachineName.ToLower(), eventLogEntry.MachineName.ToLower());
```

Comment:
It shouldn't matter here because you're doing it to both sides, but for something like MachineName, you want a culture-insensitive ToLower, so you should use ToLowerInvariant.

Reply:
I would fix these and above also